resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala

```diff
@@ -74,10 +76,12 @@ private[spark] class KubernetesSuite extends SparkFunSuite
     testBackend = IntegrationTestBackendFactory.getTestBackend
     testBackend.initialize()
     kubernetesTestComponents = new KubernetesTestComponents(testBackend.getKubernetesClient)
+    createTestSecret()
```

Comment:
I'm wondering if we should add these tests to a separate class? This file is growing in size and we can separate them out a bit for better code understanding.